test/hotspot/jtreg/containers/docker/EventGeneratorLoop.java

```diff
@@ -44,16 +46,21 @@ public static void main(String[] args) throws Exception {
             throw new IllegalArgumentException("Expecting one argument: time to run (seconds)");
         }
         int howLong = Integer.parseInt(args[0]);
+        long endTime = System.nanoTime() + TimeUnit.SECONDS.toNanos(howLong);
 
         System.out.println(MAIN_METHOD_STARTED + ", argument is " + howLong);
 
-        for (int i=0; i < howLong; i++) {
+        int count = 0;
+        while (System.nanoTime() < endTime) {
             SimpleEvent ev = new SimpleEvent();
             ev.msg = "Hello";
-            ev.count = i;
+            ev.count = count++;
             ev.commit();
```

Comment:
@kevinjwalls what do you think? I chose `System.nanoTime()` instead of `System.currentTimeMillis()` to get an even more predictable duration. 

Before this change, `EventGeneratorLoop` was guaranteed to post the same number of events as the input argument. It no longer is, but AFAICT no current test uses the events it posts either.